x.py

```diff
@@ -1,38 +1,17 @@
-#!/usr/bin/env bash
+#!/usr/bin/env python3
 
-# Modern Linux and macOS systems commonly only have a thing called `python3` and
-# not `python`, while Windows commonly does not have `python3`, so we cannot
-# directly use python in the shebang and have it consistently work. Instead we
-# embed some bash to look for a python to run the rest of the script.
-#
-# On Windows, `py -3` sometimes works. We need to try it first because `python3`
-# sometimes tries to launch the app store on Windows.
-'''':
-for PYTHON in "py -3" python3 python python2; do
-    if command -v $PYTHON >/dev/null; then
-        exec $PYTHON "$0" "$@"
-        break
-    fi
-done
-echo "$0: error: did not find python installed" >&2
-exit 1
-'''
-
-# The rest of this file is Python.
-#
 # This file is only a "symlink" to bootstrap.py, all logic should go there.
 
 import os
 import sys
 
 # If this is python2, check if python3 is available and re-execute with that
 # interpreter.
-#
-# `./x.py` would not normally benefit from this because the bash above tries
```

Comment:
Some version of this comment is still useful to acknowledge we're asking for python3 and then checking if we're actually python2

Reply:
it's right there on the line above
https://github.com/rust-lang/rust/blob/22bbacb001bc0af4b35ea167c5c1531438cc4538/x.py#L8-L9

Reply:
The difference is that now we're asking for `python3`; the point is to answer how we got here with python2. The same question as how we got here with python2 when the bash prefix should've ensured we were on python3.

Perhaps this can be implicitly understood as "how shebangs work," but then the comment wouldn't've been necessary for a bash shebang either, right?